### PR TITLE
Make NettyDockerCmdExecFactory has compatibility both Linux and OSX automatically

### PR DESCRIPTION
**Pull request by xuchenCN**
This PR make compatibility  both Linux and OSX and automatically setting the EventLoopGroup (EpollEventLoopGroup or KQueueEventLoopGroup) .

**Due to  netty/netty#6837 this PR only work on OSX 10.12**

Inspired by @marcuslinke 's branch **netty-native-kqueue** and we discussion #859 .

---
This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/862)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -259,6 +259,18 @@
 			<classifier>linux-x86_64</classifier>
 		</dependency>
 		<dependency>
+			<groupId>io.netty</groupId>
+			<artifactId>netty-transport-native-kqueue</artifactId>
+			<version>${netty.version}</version>
+			<classifier>osx-x86_64</classifier>
+			<exclusions>
```
Comment (marcuslinke) [inline pom.xml]: Why is this exclusion?

**Comment by marcuslinke** [comment]
For discussion: As netty/netty#6837 wont be fixed by the netty team this PR will work with MacOS Sierra+ only. Not sure if this is a blocker as the JAXRS implementation doesn't have this limitation.

File: pom.xml (hunk 1 above)
Comment (xuchenCN) [inline pom.xml]: Oh, I will remove this exclusion later

**Comment by KostyaSha** [comment]
@xuchenCN please squash commits

**Comment by xuchenCN** [comment]
Hi @KostyaSha sorry for late, already squashed commits. Thx

**Pull request merged**